code/obj/submachine/robotics.dm

```diff
@@ -271,26 +271,63 @@
 	icon = 'icons/obj/items/device.dmi'
 	icon_state = "atmosporter"
 	var/capacity = 2
-
-	attack_self(var/mob/user as mob)
-		if (length(src.contents) == 0) boutput(user, SPAN_ALERT("You have nothing stored!"))
-		else
-			if (user.loc != get_turf(user.loc))
-				boutput(user, SPAN_ALERT("You're in too small a space to drop anything!"))
+	var/opmode = 1 // operating mode. 1 will be pickup and 2 will be interact (drop, attack)
+
+	attack_self(var/mob/user as mob) // Switch our operating mode here
+		switch (src.opmode)
+			if (1)
+				src.opmode = 2
+				user.show_text("Operating mode set to Dropoff / Interact.", "red")
+			if (2)
+				src.opmode = 1
+				user.show_text("Operating mode set to Pick-up", "green")
+
+	afterattack(atom/A, mob/user as mob) // I have no idea what I am doing but let's go for it.
+		if (!A) return // I assume this is so we don't try and interact with a non-existent atom.
+		// Time to consider our operating mode.
+		if (BOUNDS_DIST(get_turf(src), get_turf(A)) > 0) // If it is too far away do nothing.
+			return
+		if (src.opmode == 1) // We are in pickup mode, no reason to do anything.
+			return
```

Comment:
it feels weird that the behavior for pickup and interact modes are so separated, while each implementation has an early return for the other mode. however, it just feels like a code smell; is there a reason behind this I'm missing?

Reply:
If you have, say, an air tank (normal human wearable air tank) and you want to attach it to a high capacity air cannister, you do not want to have it pick up the cannister with the attack.

Reply:
No wait a minute that is backwards. Why did I do this again?

Reply:
Going to file this under "security-blanket" and call it a day. It isn't hurting anything by existing.

Reply:
Going to file this under "i think i made obsoleted this".